cmd/overlay/cache.go

```diff
@@ -18,9 +21,21 @@ import (
 type cacheConfig struct {
 	NodesPath   string `help:"the path to a JSON file containing an object with IP keys and nodeID values"`
 	DatabaseURL string `help:"the database connection string to use"`
+	StatDBPort  string `help:"the statdb connection port to use" default:":7778"`
+	StatDBKey   string `help:"the statdb api key to use"`
 }
 
 func (c cacheConfig) open() (*overlay.Cache, error) {
+	// TODO(moby) what identity to use for statdb client?
+	identity, err := getNewIdentity()
+	if err != nil {
+		return nil, err
+	}
+	statdb, err := sdbclient.NewClient(identity, c.StatDBPort, []byte(c.StatDBKey))
```

Comment:
Is there a reason it's not directly accessing statdb instead of doing it through protobuf?

Reply:
The decision was mostly because the sdbclient allows us to abstract some of the details of interfacing with the statdb server away. For example, the sdbclient has an `UpdateUptime()` function that calls `statdb.Update()` with the correct `UpdateRequest` protobuf for updating only uptime. 

The statdb server also requires every GRPC request to have the api key attached. When creating the client, the API key is passed into the constructor, and the client deals with putting the API key in each request.

Honestly, I will probably need to refactor parts of both the server and client in the future, but I think using the client is fine until then.

Reply:
Ah, ok. Just trying to make sure that we are adding complexity when we actually need it.